### ISSUE TITLE
Feature Request: Allow puter apps to launch on other domains 

### ISSUE THREAD

**Issue by triptych** [HeyPuter/puter#246]
Right now when you call puter.ui.launchApp()
it only works when you are on puter.com GUI. This request is to allow puter.com apps to open a window and launch on any site. Just like how you can trigger the sign in dialog on any site, this would allow any site to host puter.com apps.

**Comment by AtkinsSJ** [comment]
Sounds pretty reasonable to me!

Assuming that coolapp.com calls `puter.ui.launchApp('thing')`, I can think of a couple of possibilities for how this could work:

1. A new browser tab gets opened with `puter.com/app/thing`. Very easy, doesn't even have to know if the app exists or not. But launching the whole GUI feels a bit much.
2. A new browser tab gets opened just with the app itself - what would be in the iframe if you launched it on Puter. Should be possible, I think we do expose an API for looking up app Metadata like that. That's assuming that all apps work fine when outside of Puter itself but I think that should be true?
3. Same as 2 but we launch it in a popup window instead. That might feel nicer, depending on the app.

**Comment by triptych** [comment]
Yeah I think right now anything would be better than just the error that gets thrown when trying to call the api outside the GUI

**Comment by dagelf** [comment]
Might be a way to curb expense of API fees on some of their APIs
